### ISSUE TITLE
unresolved function SSL_get_peer_certificate

### ISSUE THREAD

**Issue by Lem** [SevenTV/chatterino7#161]
### Checklist

- [X] I'm reporting a problem with Chatterino
- [X] I've verified that I'm running **the most** recent nightly build or stable release
- [X] I've looked for my problem on the [wiki](https://wiki.chatterino.com/Help/)
- [X] I've searched the [issues and pull requests](https://github.com/Chatterino/chatterino2/issues?q=) for similar looking reports

### Describe your issue

A loot of features does not work because of broken TLS/SSL. For example emotes does not load, history service failed, the red indicator is not shown. The only thing which seems to work is the chat.

In terminal I get a lot of these messages:

```
qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate
```

This issue seems to be specific to this fork as the original Chatterino 2.4.0 does not have this effect for me.
Edit: I'm using AppImage for both versions of Chatterino.

Sadly I don't know what version-information you would need for that, but I'm happy to provide them if requested.

### OS and Chatterino Version

Chatterino 7.4.0 (commit 8f57d63f modified) built with Qt 5.15.2 Running on Arch Linux, kernel: 6.0.11-arch1-1

**Comment by Felanbird** [comment]
Could you try the AppImage from the [nightly build](https://github.com/SevenTV/chatterino7/releases/tag/nightly-build) to see if the issue persists? 

Some googling around makes me think something happened during the compiling of the AppImage from the 7.4.0 release, so I'd like a quick 👀 to see if the nightly build has this same issue.

**Comment by Lem** [comment]
Sadly I've got still the same errors with that linked nightly.

From about-dialog: 
Chatterino 7.4.0 (commit 18e13277 modified) built with Qt 5.15.2
Running on Arch Linux, kernel: 6.0.11-arch1-1

**Comment by pajlada** [comment]
Can you try installing the version from AUR with an AUR helper? (e.g. `yay -S chatterino2-git`)

**Comment by jszeluga** [comment]
I was having this issue too. The new nightly app image build works. It just needed to be rebuilt.

**Comment by Nerixyz** [comment]
Is this still an issue? There have been some changes to how the AppImage is built - e.g. it's not built on Ubuntu 22.04.

**Comment by jszeluga** [comment]
I switched to the Chatterino2 flatpak and haven't looked back, but since you responded I decided to download the nightly AppImage and try it and I'm still getting the same error. I'm using Arch, if that matters at all.

`qt.network.ssl: QSslSocket: cannot resolve EVP_PKEY_base_id qt.network.ssl: QSslSocket: cannot resolve SSL_get_peer_certificate qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate chatterino.twitch: Fetching blocks failed: SslHandshakeFailedError chatterino.seventv: Couldn't load 7TV global emotes "" QTextCursor::setPosition: Position '1' out of range qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate qt.network.ssl: QSslSocket: cannot call unresolved function EVP_PKEY_base_id qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate qt.network.ssl: QSslSocket: cannot call unresolved function EVP_PKEY_base_id qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate chatterino.ivr: Failed IVR API Call! "SslHandshakeFailedError" "" qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate chatterino.bttv: Error fetching BTTV emotes for channel "212830473" , error "SslHandshakeFailedError" qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate chatterino.ffzemotes: Error fetching FFZ emotes for channel "212830473" , error "SslHandshakeFailedError" qt.network.ssl: QSslSocket: cannot call unresolved function SSL_get_peer_certificate chatterino.seventv: Error fetching 7TV emotes for channel "212830473" , error "SslHandshakeFailedError" chatterino.seventv: Error fetching 7TV emotes for channel "603776329" , error "SslHandshakeFailedError" `

**Comment by jszeluga** [comment]
> Is this still an issue? There have been some changes to how the AppImage is built - e.g. it's not built on Ubuntu 22.04.

It's not on Ubuntu 22.04 for Chatterino2, but for Chatterino7 it looks like the AppImage is built on Ubuntu 22.04 and that uses OpenSSL 3.x which is causing the problem. I was able to play around and build OpenSSL 1.1 on 22.04 and then link Chatterino7 to that and create a working AppImage. I'm not sure if you want to go that route or just wait until the switch to Qt 6

**Comment by Nerixyz** [comment]
Only 20.04 has OpenSSL 1.1 available and Qt 6.5 and up should be linked against OpenSSL 3 (maybe even Qt 6.2, not sure). So I'd wait until the full migration to Qt 6. If it remains an issue, then we'll just build OpenSSL like we do on Windows. Conan will be used on all platforms to build a static libavif anyway.
